app/models/comment.rb

```diff
@@ -46,14 +48,17 @@ class Comment < ActiveRecord::Base
     participation.unparticipate! if participation.present?
   end
 
-  def message
-    @message ||= Diaspora::MessageRenderer.new text
-  end
-
   def text= text
      self[:text] = text.to_s.strip #to_s if for nil, for whatever reason
   end
 
+  def people_allowed_to_be_mentioned
+    return @people_allowed_to_be_mentioned unless @people_allowed_to_be_mentioned.nil?
+    contacts = author.owner.try(:contact_people) if parent.public?
+
+    @people_allowed_to_be_mentioned = parent.try(:participants).to_a.concat(contacts.to_a).push(parent.author).uniq
```

Comment:
> I'm not quite sure I follow why we know parent.author always succeeds but parent.participants may not

We have a [spec](https://github.com/diaspora/diaspora/blob/develop/spec/models/photo_spec.rb#L209) for photos commenting, and photos don't have participants. This spec fails when I do `parent.participants` with no `try`. Am I right that we don't have separate photos comments allowed now?